MDANSE/Src/MDANSE/Framework/Configurators/GroupingLevelConfigurator.py

```diff
@@ -92,49 +142,280 @@ def configure(self, value):
                     elements.append([chemical_system.atom_list[x] for x in cluster])
                     names.append(f"{mol_name}_mol{mol_number + 1}")
                     masses.append([mass_lookup[x] for x in cluster])
-        elif value == "group":
-            for group_name, group_indices in chemical_system._labels.items():
-                residue = set(group_indices)
-                counter = 1
-                for clustername, clusterlist in chemical_system._clusters.items():
-                    for cluster in clusterlist:
-                        molecule = set(cluster)
-                        if molecule.issubset(residue) or residue.issubset(molecule):
-                            indices.append(list(molecule.intersection(residue)))
-                            elements.append(
-                                [chemical_system.atom_list[x] for x in cluster]
-                            )
-                            names.append(f"{group_name}_num{counter}_in_{clustername}")
-                            masses.append([mass_lookup[x] for x in cluster])
-                            counter += 1
 
         atomSelectionConfig["indices"] = indices
+        atomSelectionConfig["flatten_indices"] = collapse(indices)
         atomSelectionConfig["elements"] = elements
         atomSelectionConfig["masses"] = masses
         atomSelectionConfig["names"] = names
         atomSelectionConfig["selection_length"] = len(names)
-        atomSelectionConfig["unique_names"] = sorted(set(atomSelectionConfig["names"]))
+        atomSelectionConfig["unique_names"] = sorted(set(names))
 
-        self["level"] = value
-        self["group_indices"] = list(range(len(names)))
         if atomSelectionConfig["selection_length"] == 0:
             self.error_status = "This option resulted in nothing being selected in the current trajectory"
 
-    @staticmethod
-    def find_parent(atom, level):
+    def get_element_from_label(self, label: str) -> str:
+        """Returns the element for a given label.
+
+        Parameters
+        ----------
+        label : str
+            The label of the element e.g. [H2_O1]_H
+
+        Returns
+        -------
+        str
+            The element of the inputted label.
         """
-        Retrieve recursively the parent of a given atom at a given level.
-        For example, a level of 1 will return the direct parent of the atom.
+        if self["level"] == "atom":
+            return label
+        return self["name_to_element"][label]
 
-        :note: this is a static method
+    def add_grouped_totals(
+        self,
+        output_data: dict[str, npt.NDArray],
+        result_name: str,
+        data_type: str,
+        dim: int = 1,
+        conc_exp: float = 1.0,
+        intra: bool = False,
```

Comment:
`bool`s and large argument counts of extra arguments are usually best expressed as kwarg only. 
```suggestion
        *,
        intra: bool = False,
```